internal/parser/simple_parser_rules.go

```diff
@@ -929,3 +933,157 @@ func (p *simpleParser) parseRollbackStmt(r reporter) (stmt *ast.RollbackStmt) {
+// https://sqlite.org/lang_createindex.html
+func (p *simpleParser) parseCreateIndexStmt(r reporter) (stmt *ast.CreateIndexStmt) {
+	stmt = &ast.CreateIndexStmt{}
+	p.searchNext(r, token.KeywordCreate)
+	next, ok := p.lookahead(r)
+	if !ok {
+		return
+	}
+	stmt.Create = next
+	p.consumeToken()
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Type() == token.KeywordUnique {
+		stmt.Unique = next
+		p.consumeToken()
+	}
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Type() == token.KeywordIndex {
+		stmt.Index = next
+		p.consumeToken()
+	}
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Type() == token.KeywordIf {
+		stmt.If = next
+		p.consumeToken()
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		if next.Type() == token.KeywordNot {
+			stmt.Not = next
+			p.consumeToken()
+			next, ok = p.lookahead(r)
+			if !ok {
+				return
+			}
+			if next.Type() == token.KeywordExists {
+				stmt.Exists = next
+				p.consumeToken()
+			} else {
+				r.unexpectedToken(token.KeywordExists)
+			}
+		} else {
+			r.unexpectedToken(token.KeywordNot)
+		}
+	}
+
+	schemaNameOrIndexName, ok := p.lookahead(r)
+	if !ok {
+		return
+	}
+	if schemaNameOrIndexName.Type() == token.Literal {
+		// This is the case where there might not be a schemaName
+		// We assume that its the table name in the beginning and assign it
+		stmt.IndexName = schemaNameOrIndexName
+		p.consumeToken()
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		// If we find the signs of there being a SchemaName,
+		// we over-write the previous value
+		if next.Value() == "." {
+			stmt.SchemaName = schemaNameOrIndexName
+			stmt.Period = next
+			p.consumeToken()
+			indexName, ok := p.lookahead(r)
+			if !ok {
+				return
+			}
+			stmt.IndexName = indexName
+			p.consumeToken()
+		}
+	}
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Type() == token.KeywordOn {
+		stmt.On = next
+		p.consumeToken()
+	}
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Type() == token.Literal {
+		stmt.TableName = next
+		p.consumeToken()
+	}
+
+	next, ok = p.lookahead(r)
+	if !ok {
+		return
+	}
+	if next.Value() == "(" {
+		stmt.LeftParen = next
+		p.consumeToken()
+		next, ok = p.lookahead(r)
+		if !ok {
+			return
+		}
+		for next.Value() != ")" {
+			stmt.IndexedColumns = append(stmt.IndexedColumns, p.parseIndexedColumns(r))
+			next, ok = p.lookahead(r)
+			if !ok {
+				return
+			}
+			if next.Value() == "," {
+				p.consumeToken()
+			}
+			next, ok = p.lookahead(r)
+			if !ok {
+				return
+			}
+		}
+		if next.Value() == ")" {
+			stmt.RightParen = next
+			p.consumeToken()
+		}
+	}
+
+	next, ok = p.optionalLookahead(r)
+	if !ok || next.Type() == token.EOF {
+		return
+	}
+	if next.Type() == token.KeywordWhere {
+		stmt.Where = next
+		p.consumeToken()
+		stmt.Expr = p.parseExpression(r)
+	}
+	return
+}
+
+func (p *simpleParser) parseIndexedColumns(r reporter) (stmt *ast.IndexedColumn) {
```

Comment:
Does it really parse many?

Reply:
You mean separated by commas? Probably not, still a WIP

Reply:
This parses one single Indexed column and is appended later above. Ill change the name to `parseIndexedColumn`